10.1/root/usr/share/container-scripts/mysql/check-upgrade.sh

```diff
@@ -0,0 +1,47 @@
+check_datadir_version() {
+  local datadir="$1"
+  local datadir_version=$(get_datadir_version "$datadir")
+  local mysqld_version=$(mysqld_compat_version)
+
+  case "${MYSQL_UPGRADE}" in
+    auto|warn)
+      if [ "${datadir_version}" -eq "${mysqld_version}" ] ; then
+        log_info 'MySQL server version check OK.'
+        return 0
+      fi
+
+      if [ $(( ${datadir_version} + 1 )) -eq "${mysqld_version}" -o "${datadir_version}" -eq 505 -a "${mysqld_version}" -eq 1000 ] ; then
+        log_info "MySQL server is version '${mysqld_version}' and datadir is version '${datadir_version}'."
+        if [ "${MYSQL_UPGRADE}" == 'auto' ] ; then
+          log_info 'Running mysql_upgrade ...'
+          mysql_upgrade --socket=/tmp/mysql.sock
+        fi
+      else
+        log_info "MySQL server is version '${mysqld_version}' and datadir is version '${datadir_version}', which are incompatible."
```

Comment:
Also, maybe we should info user about the possibilities (what is the name of the container that supports the upgrade from this datadir version, and that multiple upgrade steps might be needed e.g.).